app/controllers/concerns/phone_confirmation_flow.rb

```diff
@@ -72,11 +72,10 @@ def send_confirmation_code
     # user's session. Re-sending the confirmation code doesn't generate a new one.
     self.confirmation_code = generate_confirmation_code unless confirmation_code
 
-    if current_otp_method == :voice
-      VoiceSenderOtpJob.perform_later(confirmation_code, unconfirmed_phone)
-    else
-      SmsSenderOtpJob.perform_later(confirmation_code, unconfirmed_phone)
-    end
+    job = "#{current_otp_method.to_s.capitalize}SenderOtpJob".constantize
+
+    job.perform_later(confirmation_code, unconfirmed_phone)
+
```

Comment:
This change adheres to the Open/Closed principle because it allows us to add new delivery options without changing the controller code, whereas before, we would need to add a new `elsif` statement to the conditional in the controller.

Here's a quick article on SOLID principles: https://robots.thoughtbot.com/back-to-basics-solid

I also highly recommend Sandi Metz's *Practical Object-Oriented Design in Ruby: An Agile Primer*, which is available on Safari Books Online.